src/components/actions/coreActions.ts

```diff
@@ -0,0 +1,239 @@
+/*
+Contains core actions that must be run for the system to function correctly.
+These should NEVER be configurable or removable by template configurations, so
+they've been hard-coded here in order to:
+- be maintained in one place: a change can be made without having to
+  re-configure all templates
+- prevent accidental misconfiguration or removal
+*/
+
+import { Trigger } from '../../generated/graphql'
+import { ActionInTemplate } from '../../types'
+
+type CoreActions = {
+  [key in Trigger]?: Omit<ActionInTemplate, 'parameters_evaluated'>[]
+}
+
+const coreActions: CoreActions = {
+  [Trigger.OnApplicationCreate]: [
+    // Generate serial for application when first created
+    // Can be over-ridden by specifying a template action
+    {
+      code: 'generateTextString',
+      path: '../plugins/action_generate_text_string/src/index.ts',
+      name: 'Generate Text String',
+      trigger: 'ON_APPLICATION_CREATE',
+      event_code: null,
+      sequence: -3,
+      condition: true,
+      parameter_queries: {
+        pattern: 'S-[A-Z]{3}-<+dddd>',
```

Comment:
To-Do: fetch this pattern from template config

Reply:
Is this included on the next PR: #978 ?

Reply:
Yes